### PR TITLE
samples: add create agent code sample

### PR DESCRIPTION
**Pull request by galdawave**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-dialogflow-cx/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

### CODE REVIEW

**Review by Neenu1995 — APPROVED** [review]

File: samples/snippets/src/main/java/dialogflow/cx/CreateAgent.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import com.google.cloud.dialogflow.cx.v3.Agent;
+import com.google.cloud.dialogflow.cx.v3.Agent.Builder;
+import com.google.cloud.dialogflow.cx.v3.AgentsClient;
+import com.google.cloud.dialogflow.cx.v3.AgentsSettings;
+import java.io.IOException;
+
+public class CreateAgent {
+
+  public static void main(String[] args) throws IOException {
+    createAgent("My-Project-Id", "Agent-Name");
```
Comment (telpirion) [inline samples/snippets/src/main/java/dialogflow/cx/CreateAgent.java:30]: NIT: I think a common format here is to declare these as `String` variables in `main()` and then pass the variables into the call to the method (`createAgent()`) from within `main()`.

File: samples/snippets/src/main/java/dialogflow/cx/CreateAgent.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    Builder build = Agent.newBuilder();
+    build.setDefaultLanguageCode("en");
+    build.setDisplayName(agentName);
+    build.setTimeZone("America/Los_Angeles");
+
+    Agent agent = build.build();
+    String parentPath = "projects/" + parent + "/locations/global";
+
+    AgentsSettings agentsSettings = AgentsSettings.newBuilder().setEndpoint(apiEndpoint).build();
+    AgentsClient client = AgentsClient.create(agentsSettings);
```
Comment (telpirion) [inline samples/snippets/src/main/java/dialogflow/cx/CreateAgent.java:47]: Move this up towards the top of the sample and place within a `try` block.

The first few lines of this sample should looks something like this:

```
String apiEndpoint = "global-dialogflow.googleapis.com:443";
AgentsSettings agentsSettings = AgentsSettings.newBuilder()
    .setEndpoint(apiEndpoint)
    .build();
try (AgentsClient client = AgentsClient.create(agentsSettings)) {

// ... rest of sample
```

File: samples/snippets/src/main/java/dialogflow/cx/CreateAgent.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    // Set the details of the Agent to create
+    Builder build = Agent.newBuilder();
+    build.setDefaultLanguageCode("en");
+    build.setDisplayName(agentName);
+    build.setTimeZone("America/Los_Angeles");
+
+    Agent agent = build.build();
+    String parentPath = "projects/" + parent + "/locations/global";
+
+    AgentsSettings agentsSettings = AgentsSettings.newBuilder().setEndpoint(apiEndpoint).build();
```
Comment (telpirion) [inline samples/snippets/src/main/java/dialogflow/cx/CreateAgent.java:46]: Insert a linebreak after call to `newBuilder()`.

File: samples/snippets/src/main/java/dialogflow/cx/CreateAgent.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+  public static Agent createAgent(String parent, String agentName) throws IOException {
+
+    String apiEndpoint = "global-dialogflow.googleapis.com:443";
+
+    // Set the details of the Agent to create
+    Builder build = Agent.newBuilder();
+    build.setDefaultLanguageCode("en");
+    build.setDisplayName(agentName);
+    build.setTimeZone("America/Los_Angeles");
```
Comment (telpirion) [inline samples/snippets/src/main/java/dialogflow/cx/CreateAgent.java:41]: NIT: Might want to include a comment that describes the correct protocol/format for TimeZone strings.

**Review by telpirion — APPROVED** [review]

**Comment by telpirion** [comment]
Adding other java-samples-reviewers for input.

File: samples/snippets/src/main/java/dialogflow/cx/CreateAgent.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+      Builder build = Agent.newBuilder();
+
+      build.setDefaultLanguageCode("en");
+      build.setDisplayName(displayName);
+      //Correct format for timezone is location/city
+      //For example America/Los_Angeles, Europe/Madrid, Asia/Tokyo
+      build.setTimeZone("America/Los_Angeles");
+
+      Agent agent = build.build();
+      String parentPath = "projects/" + parent + "/locations/global";
```
Comment (lesv) [inline samples/snippets/src/main/java/dialogflow/cx/CreateAgent.java:53]: Ok, but in the future please prefer `String.format("projects/%s/locations/%s", parent, "global");`

File: samples/snippets/src/test/java/dialogflow/cx/CreateAgentIT.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    System.setOut(new PrintStream(stdOut));
+
+  }
+
+  @After
+  public void tearDown() throws IOException {
+    stdOut = null;
+    System.setOut(null);
+    String apiEndpoint = "global-dialogflow.googleapis.com:443";
+    String parentPath = "projects/" + PROJECT_ID + "/locations/global";
```
Comment (lesv) [inline samples/snippets/src/test/java/dialogflow/cx/CreateAgentIT.java:49]: Ok here, but it's a good habit to get into to use `String.format`

**Review by lesv — APPROVED** [review]

**Review by lesv — APPROVED** [review]

File: samples/snippets/src/main/java/dialogflow/cx/CreateAgent.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+import com.google.cloud.dialogflow.cx.v3.Agent;
+import com.google.cloud.dialogflow.cx.v3.Agent.Builder;
+import com.google.cloud.dialogflow.cx.v3.AgentsClient;
+import com.google.cloud.dialogflow.cx.v3.AgentsSettings;
+import java.io.IOException;
+
+public class CreateAgent {
+
+  public static void main(String[] args) throws IOException {
+
```
Comment (averikitsch) [inline samples/snippets/src/main/java/dialogflow/cx/CreateAgent.java:30]: nit: add todo to replace variables

File: samples/snippets/src/main/java/dialogflow/cx/CreateAgent.java [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+      // Correct format for timezone is location/city
+      // For example America/Los_Angeles, Europe/Madrid, Asia/Tokyo
+      build.setTimeZone("America/Los_Angeles");
+
+      Agent agent = build.build();
+      String parentPath = String.format("projects/%s/locations/%s", parent, "global");
+
+      // Make API request to create agent
+      Agent response = client.createAgent(parentPath, agent);
+      System.out.println(response);
```
Comment (averikitsch) [inline samples/snippets/src/main/java/dialogflow/cx/CreateAgent.java:57]: nit: is there anything specific you want to highlight about the response?

File: samples/snippets/src/test/java/dialogflow/cx/CreateAgentIT.java [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+  @Before
+  public void setUp() throws IOException {
+    stdOut = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(stdOut));
+  }
+
+  @After
+  public void tearDown() throws IOException {
+    stdOut = null;
+    System.setOut(null);
```
Comment (averikitsch) [inline samples/snippets/src/test/java/dialogflow/cx/CreateAgentIT.java:46]: We should restore to the original

**Review by averikitsch — APPROVED** [review]

File: samples/snippets/src/main/java/dialogflow/cx/CreateAgent.java (hunk 5 above)
Comment (averikitsch) [reply samples/snippets/src/main/java/dialogflow/cx/CreateAgent.java]: I thought we were advocating for the addition not the string.format() due to performance?

**Review by stephaniewang526 — APPROVED** [review]

**Pull request merged**